core/sail/inferencer/src/main/java/org/eclipse/rdf4j/sail/inferencer/fc/SchemaCachingRDFSInferencer.java

```diff
@@ -636,4 +636,8 @@ public boolean isAddInferredStatementsToDefaultContext() {
 	public void setAddInferredStatementsToDefaultContext(boolean addInferredStatementsToDefaultContext) {
 		this.addInferredStatementsToDefaultContext = addInferredStatementsToDefaultContext;
 	}
+
+	boolean usesPredefinedSchema() {
+		return predefinedSchema != null || sharedSchema;
+	}
```

Comment:
This is the key change here. I use a somewhat undocumented part of the RDFS reasoner that allows you to create a base reasoner that can be used to instantiate a new one much faster than building one from scratch by loading in the schema every time. There was a bug that caused the reasoner to think that new schema data was added, making everything fairly slow. Fixed that.